Use Host Architecture as Fallback Architecture in JavascriptCore Tests

chgibb-apple opened the pull request:
#### c7b924825a747a3526b8734c6bc0c6e7377705e8
<pre>
Use Host Architecture as Fallback Architecture in JavascriptCore Tests
<a href="https://bugs.webkit.org/show_bug.cgi?id=265681">https://bugs.webkit.org/show_bug.cgi?id=265681</a>
<a href="https://rdar.apple.com/119040784">rdar://119040784</a>

Reviewed by Jonathan Bedard and Alexey Proskuryakov.

Allow for running JSC stress tests on macOS hosts that do not have Xcode installed.

* Tools/Scripts/run-javascriptcore-tests:

Canonical link: <a href="https://commits.webkit.org/271513@main">https://commits.webkit.org/271513@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a868be216dbf88844d26c1986865717350776eb9

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/28591 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/7236 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/29974 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/31239 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/26115 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/29097 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/9414 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/4606 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/31239 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/28861 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/5982 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/24585 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/31239 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/5354 "Unexpected infrastructure issue, retrying build") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/25586 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/32556 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/24740 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/26182 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/26031 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/31604 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/27679 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/5318 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/3488 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/29388 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/6923 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/35209 "Built successfully") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/5779 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/7606 "Passed tests") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3696 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/5837 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Tools/Scripts/run-javascriptcore-tests:
```diff
@@ -46,6 +46,9 @@ use webkitdirs;
 use Scalar::Util qw(looks_like_number);
 use Text::ParseWords;
 
+sub trim { my $s = shift; $s =~ s/^\s+|\s+$//g; return $s };
```
We always use `chomp` for this, no need to introduce a new function.

chgibb-apple replied:
Fixed.

aproskuryakov commented on Tools/Scripts/run-javascriptcore-tests:
```diff
@@ -555,8 +558,8 @@ if (defined $remoteConfigFile) {
 } else {
-    # Fallback is x86_64, which we replace with the native architecture if the native architecture is in the provided build
-    $archs = "x86_64";
+    # Fallback is the architecture of the host machine
```
This comment says what the code below does, not why it does that. Please add some discussion of how arm64 vs. arm64e is handled in this case. We always have problems with that. 

Also, please add a period at the end of the comment.

chgibb-apple replied:
arm64e is handled here https://github.com/WebKit/WebKit/blob/main/Tools/Scripts/run-javascriptcore-tests#L564 (just below my changes). I've added a note in this comment about using `/usr/bin/arch` in favour of `lipo` to avoid the Xcode dependency.

aproskuryakov commented on Tools/Scripts/run-javascriptcore-tests:
```diff
@@ -46,6 +46,7 @@ use webkitdirs;
 use Scalar::Util qw(looks_like_number);
 use Text::ParseWords;
 
+
```
Extra line added.

chgibb-apple replied:
Fixed.

aproskuryakov commented on Tools/Scripts/run-javascriptcore-tests:
```diff
@@ -555,8 +556,9 @@ if (defined $remoteConfigFile) {
-    $archs = "x86_64";
+    # Fallback is the architecture of the host machine. Avoid calling lipo which allows this script to work without Xcode being installed.
+    $archs = trim(`/usr/bin/arch`);
+    chomp($archs);
```
Should be:
```
chomp($archs = `/usr/bin/arch`);
```

chgibb-apple replied:
Thank you for catching. Fixed.

chgibb-apple committed:
Use Host Architecture as Fallback Architecture in JavascriptCore Tests
https://bugs.webkit.org/show_bug.cgi?id=265681
rdar://119040784

Reviewed by Jonathan Bedard and Alexey Proskuryakov.

Allow for running JSC stress tests on macOS hosts that do not have Xcode installed.

* Tools/Scripts/run-javascriptcore-tests:

Canonical link: https://commits.webkit.org/271513@main
